SPECIFICATION.md

```diff
@@ -0,0 +1,177 @@
+
+Example:
+
+    [format]
+    format = FontoBene
+    format_version = 1.0
+
+    [font]
+    name = LibrePCB Font
+    id = librepcb
+    author = Max Müller <max@example.com>
+    version = 1.0
+    license = GPL-3.0+
+
+    [user]
+    comment = This is the official font used for LibrePCB projects.
+
+The valid keys are standardized in the "Formal Specification" section.
+
+
+### Body
+
+The first empty line separates the header from the body. The body consists of
+glyph definition blocks separated by at least one empty line.
+
+Every glyph definition starts with the glyph declaration, a unicode codepoint
+in square brackets (e.g. `[1F4A9]` for the pile of poo character). It may
+optionally be followed by a UTF-8 representation of the character for easier
+previewing (e.g. `💩`). However, that preview will be ignored by the parser.
+
+The notation used inside the brackets follows the convention described in [*The
+Unicode Standard, Version 10.0, Appendix A, Notational Conventions, Code
+Points*][unicode-10-spec]:
+
+> In running text, an individual Unicode code point is expressed as U+n, where
+> n is four to six hexadecimal digits, using the digits 0–9 and uppercase
+> letters A–F (for 10 through 15, respectively).
+>
+> Leading zeros are omitted, unless the code point would have fewer than four
+> hexadecimal digits—for example, U+0001, U+0012, U+0123, U+1234, U+12345,
+> U+102345.
+>
+> The U+ may be omitted for brevity in tables or when denoting ranges.
+
+The glyph declaration is followed by a list of polylines or glyph references,
+each on a single line.
+
+#### Polylines
+
+A polyline looks like this:
+
+    0,0;3,9;6,0
+
+It consists of coordinate pairs, separated by semicolons (`;`). The X and Y
+coordinates are separated with a comma (`,`).  The coordinates are arbitrary
+precision floating point numbers using the full stop (U+002E) as decimal mark.
+If a coordinate is a whole number, the decimal mark and the fractional part of
+the number may be omitted.
+
+The coordinate system starts at the lower left corner. Positive X coordinates
+point to the right. Positive Y coordinates point to the top. Negative
+coordinates are not allowed.
+
+          Y
+    (0,Y) +
+          |
+          |
+          +-----+ X
+      (0,0)   (X,0)
+
+The example above would draw the pointy "outer frame" of the character "A".
+Here is the complete definition of the glyph for the character "LATIN CAPITAL
+LETTER A (U+0041)":
+
+    [0041] A
+    0.8333,2.5;5.1666,2.5
+    0,0;3,9;6,0
+
+#### References
+
+Because many glyphs are very similar, references may be used as "includes".
+References start with the "C" symbol (U+0043), followed by a codepoint. Here is
+the definition of the glyph for the character "LATIN CAPITAL LETTER A WITH
+CIRCUMFLEX (U+00C2)":
+
+    [00c2] Â
+    C0041
+    1.5,11.5;3,13;4.5,11.5
+
+An implementation would simply replace the "C0041" part with the expanded
+polylines from the referenced glyph.
```

Comment:
Should we add a possibility to define meta-glyphs that are only used for includes, but are not valid glyphs themselves? The circumflex is a bad example because it actually is a valid character, but there could be others.

Also, should we make it possible to translate and/or scale includes? E.g. with `C0041 offset=0.5,1.5 scale=1.2`